fastf1/core.py

```diff
@@ -2844,18 +2844,32 @@ def pick_fastest(self, only_by_time: bool = False) -> "Lap":
         Returns:
             instance of :class:`Lap`
         """
+        # TODO: Deprecate returning empty lap object when there is no lap
+        # that matches definion
         if only_by_time:
             laps = self  # all laps
         else:
             # select only laps marked as personal fastest
-            laps = self.loc[self['IsPersonalBest'] == True]  # noqa: E712 comparison with True
+            laps = self.loc[self['IsPersonalBest'] == True]  # noqa: E712
 
         if not laps.size:
+            warnings.warn(("None will be returned instead of an empty Lap "
+                           "object when there are no laps with "
+                           "IsPersonalBest=True starting from version 3.3"),
```

Comment:
``"...with IsPersonalBest=True..."`` this isn't strictly true, is it? We will also return an empty Lap object now and None later if self was an empty Laps object. And that can happen easily by selecting a driver that didn't participate in the session or similar. So you don't need to limit the warning to this specific case IMO.

Reply:
As written it can be interpreted to cover the empty Laps object case as well. How would you reword it?

Reply:
Yes, but it also applies to the case where ``only_by_time=True`` and therefore ``IsPersonalBest`` doesn't matter. I might have worded my first comment a bit confusing.
None will be returned whenever there is no lap that qualifies as fastest lap. So, something along those lines. There are various cases for that, no lap time, not ``IsPersonalBest``, simply not lap at all. I don't think the should all be mentioned in the warning.

Reply:
Makes sense. I will rewrite